source/postprocess/visualization/theta_tauisa.cc

```diff
@@ -0,0 +1,215 @@
+                    double factor = 1.0;
+                    for (unsigned int j = 0; j < i; ++j)
+                      {
+                        multiplier = multiplier
+                                     * Tensor<2, dim>(velocity_gradient[0]);
+                        factor *= tmax / (j + 1);
+                      }
+                    multiplier = factor * multiplier;
+                    F += multiplier;
+                  }
+
+                // The "left stretch" tensor is computed from the deformation gradient tensor
+                // Note that Kaminski and Ribe incorrectly wrote U=Ft*F (typo, I guess).
+                Tensor<2, dim> Ft = transpose(F);
+                SymmetricTensor<2, dim> U = SymmetricTensor<2, dim>(F * Ft);
+                // The ~infinite strain axis is the first eigenvector of the "left stretch" tensor.
+                Tensor<1, dim> ehat = eigenvectors(U)[0].second;
+
+                // Calculate theta by dotting the ISA (ehat) and the velocity (scaled to a unit vector)
+                // theta is limited to be less than pi/2 (stay in the first quadrant)
+                double umag = std::sqrt(velocities[0] * velocities[0]);
+                double umagmin = 1.0e-40;  // to avoid problems with /0 in acos
+                double theta_val = 0;
+                if (umag > umagmin)
+                  {
+                    theta_val = std::acos(std::abs((velocities[0] * ehat) / umag));
+                    if (theta_val > 1.57079632679)
+                      theta_val = fmod(theta_val, 1.57079632679);
+                  }
+                (*return_value.second)(cell_index) = theta_val;
+              }
+          }
+        return return_value;
+      }
+
+      template<int dim>
+      std::pair<std::string, Vector<float> *> tauISA<dim>::execute() const
+      {
+        std::pair<std::string, Vector<float> *> return_value("tauISA",
+                                                             new Vector<float>(this->get_triangulation().n_active_cells()));
+
+        const QMidpoint<dim> quadrature_formula;
+        const unsigned int n_q_points = quadrature_formula.size();
+
+        FEValues<dim> fe_values(this->get_mapping(), this->get_fe(),
+                                quadrature_formula,
+                                update_values | update_gradients | update_quadrature_points);
+
+        // Set up material models
+        MaterialModel::MaterialModelInputs<dim> in(n_q_points,
+                                                   this->n_compositional_fields());
+        MaterialModel::MaterialModelOutputs<dim> out(n_q_points,
+                                                     this->n_compositional_fields());
+
+        // Set up cell iterator for looping
+        typename DoFHandler<dim>::active_cell_iterator cell =
+          this->get_dof_handler().begin_active(), endc =
+            this->get_dof_handler().end();
```

Comment:
here and below, may of the same comments apply as for your first postprocessor. I won't repeat them, but please go through the code and address what you see fits in the same style as my questions/comments above